### PR TITLE
Test/cover constructor initialization

### LINKED ISSUE #282: V2 Cover unit tests: Cover.sol constructor + initialization
**Constructor**
- [ ] Should set variables correctly

**Initialize**
- [ ] Should initialize variables correctly
- [ ] Should not be able to initialize twice
- [ ] Should we tests other methods to work or not if Cover not initialized?

### PR DESCRIPTION
**Pull request by paterson1**
## Context

Closes #282 

## Changes proposed in this pull request

Added tests for the constructor and initialize functions of `Cover.sol`. `setup` was changed to expose some values to make it easier to deploy a new `Cover` not initialized.

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: test/unit/Cover/initialize.js [hunk 1]
```diff
@@ -1,28 +1,60 @@
...
+    const { quotationData, stakingPool, futureCoverNFTAddress, coverAddress, coverUtilsLib } = this;
+    const Cover = await ethers.getContractFactory('Cover', {
+      libraries: {
+        CoverUtilsLib: coverUtilsLib.address,
+      },
+    });
+
+    // Uninitialized cover
+    const cover = await Cover.deploy(
+      quotationData.address,
```
Comment (danoctavian) [inline test/unit/Cover/initialize.js:15]: these values

quotationData.address
productsV1.address

are private fields. However we can try using getStorageAt https://docs.ethers.io/v5/api/providers/provider/#Provider-getStorageAt

to get the field value.

Let's try asserting them regardless using getStorageAt

File: test/unit/Cover/constructor.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    const { quotationData, stakingPool, futureCoverNFTAddress, coverAddress, coverUtilsLib } = this;
+
+    const Cover = await ethers.getContractFactory('Cover', {
+      libraries: {
+        CoverUtilsLib: coverUtilsLib.address,
+      },
+    });
+
+    const cover = await Cover.deploy(
+      quotationData.address,
```
Comment (danoctavian) [inline test/unit/Cover/constructor.js:15]: these values

quotationData.address
productsV1.address

are private fields. However we can try using getStorageAt https://docs.ethers.io/v5/api/providers/provider/#Provider-getStorageAt

to get the field value.

Let's try asserting them regardless using getStorageAt

File: test/unit/Cover/setup.js [hunk 3]
```diff
@@ -215,6 +215,11 @@ async function setup() {
   this.coverNFT = coverNFT;
   this.accounts = accounts;
   this.capacityFactor = capacityFactor;
+  this.coverUtilsLib = coverUtilsLib;
+  this.quotationData = quotationData;
+  this.stakingPool = stakingPool;
+  this.coverAddress = coverAddress;
+  this.futureCoverNFTAddress = futureCoverNFTAddress;
```
Comment (danoctavian) [inline test/unit/Cover/setup.js:222]: can add a dummy productsV1 address here.

**Review by danoctavian — CHANGES_REQUESTED** [review]
Let's try to assert private fields with getStorageAt reads and a few other things

File: test/unit/Cover/constructor.js (hunk 2 above)
Comment (paterson1) [reply test/unit/Cover/constructor.js:15]: I've tried with `hardhat-storage-layout` and couldn't find them, and then I realized that they are immutable so they are not saved to storage

**Comment by danoctavian** [comment]
Needs rebase.

**Pull request merged**